File: TypeScript/Random/random-int.ts
import * as crypto from "crypto"

function main(): void {
    // Generate a random int between 0 and max
    console.log(generateRandomInt(10))
}

main()

// Generate a random int between 0 and max
function generateRandomInt(max: number): number {
    return crypto.randomInt(max)
}

File: Go/Random/random-int.go
package main

import (
	"crypto/rand"
	"fmt"
	"log"
	"math/big"
)

func main() {
	// Generate a random int between 0 and 100
	fmt.Println(generateRandomInt(100))
}

// Generate a random int between 0 and a given max
func generateRandomInt(max int64) int {
	randomInt, err := rand.Int(rand.Reader, big.NewInt(max))
	if err != nil {
		log.Fatalln(err)
	}
	return int(randomInt.Int64())
}
